commerce-product-definitions-api/src/main/java/com/liferay/commerce/product/definitions/display/context/BaseCPDefinitionsSearchContainerDisplayContext.java

```diff
@@ -0,0 +1,301 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.definitions.display.context;
+
+import com.liferay.commerce.product.definitions.portlet.action.ActionHelper;
+import com.liferay.frontend.taglib.servlet.taglib.ManagementBarFilterItem;
+import com.liferay.portal.kernel.dao.search.EmptyOnClickRowChecker;
+import com.liferay.portal.kernel.dao.search.RowChecker;
+import com.liferay.portal.kernel.dao.search.SearchContainer;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.portlet.PortalPreferences;
+import com.liferay.portal.kernel.portlet.PortletURLUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.portlet.PortletException;
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Alessio Antonio Rendina
+ * @author Marco Leo
+ * @author Luca Pellizzon
+ */
+public abstract class BaseCPDefinitionsSearchContainerDisplayContext<T>
+	extends BaseCPDefinitionsDisplayContext {
+
+	public BaseCPDefinitionsSearchContainerDisplayContext(
+		ActionHelper actionHelper, HttpServletRequest httpServletRequest,
+		String portalPreferenceNamespace) {
+
+		super(actionHelper, httpServletRequest);
+
+		_portalPreferenceNamespace = portalPreferenceNamespace;
+
+		_defaultOrderByCol = "modified-date";
+		_defaultOrderByType = "asc";
+	}
+
+	public String getDisplayStyle() {
+		if (_displayStyle == null) {
+			_displayStyle = getDisplayStyle(
+				httpServletRequest, portalPreferences);
+		}
+
+		return _displayStyle;
+	}
+
+	public String getKeywords() {
+		if (_keywords != null) {
+			return _keywords;
+		}
+
+		_keywords = ParamUtil.getString(httpServletRequest, "keywords");
+
+		return _keywords;
+	}
+
+	public List<ManagementBarFilterItem> getManagementBarStatusFilterItems()
+		throws PortalException, PortletException {
+
+		List<ManagementBarFilterItem> managementBarFilterItems =
+			new ArrayList<>();
+
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_ANY));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_DRAFT));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_SCHEDULED));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_APPROVED));
+		managementBarFilterItems.add(
+			getManagementBarFilterItem(WorkflowConstants.STATUS_EXPIRED));
```

Comment:
If order matters here, we should add blank lines in between each `add` to make that clear